torch/nn/modules/loss.py

```diff
@@ -1121,6 +1125,91 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
                                ignore_index=self.ignore_index, reduction=self.reduction)
 
 
+class CrossEntropyLossWithSoftLabels(_Loss):
+    r"""This criterion combines :class:`~torch.nn.LogSoftmax` and :class:`~torch.nn.KLDivLoss` in
+    one single class. It is useful when training a classification problem with `C` classes when
+    "soft labels" are required, such as when utilizing label smoothing or blended labels.
+
+    The `input` is expected to contain raw, unnormalized scores for each class and the `target`
+    should match the `input` shape and contain the target distribution over the classes.
+
+    The loss can be described as:
+
+    .. math::
+        L = l_{n,c} = y_{n,c} \cdot \left( \log y_{n,c} - \\
+        \log\left(\frac{\exp(x_{n,c})}{\sum_{i=1}^C \exp(x_{n,i})} \right) \right)
+
+    where :math:`x_{n,c}` represents each element of ``input``, :math:`y_{n,c}` represents each element of
+    ``target``, and :math:`L` has the same shape as ``input`` and ``target``.  If :attr:`reduction` is not
+    ``'none'`` (default ``'mean'``), then:
+
+    .. math::
+        \ell(x, y) = \begin{cases}
```

Comment:
I know the other losses have the same notation, but it's not clear that `l(x, y)` is the output of the module. We probably want something like "If reduction is not `none`, then the result is a reduced loss `l(x, y)` such that..."